tests/keras/preprocessing/test_image.py

```diff
@@ -55,5 +55,32 @@ def test_image_data_generator():
             assert x.shape[1:] == images.shape[1:]
             break
 
+def test_img_flip():
+    x = np.array(range(4)).reshape([1,1,2,2])
+    assert (flip_axis(x, 0) == x).all()
+    assert (flip_axis(x, 1) == x).all()
+    assert (flip_axis(x, 2) == [[[[2, 3], [0, 1]]]]).all()
+    assert (flip_axis(x, 3) == [[[[1, 0], [3, 2]]]]).all()
+
+    dim_ordering_and_col_index = (('tf', 1), ('th', 2))
+    for dim_ordering, col_index in dim_ordering_and_col_index:
+        image_generator_th = ImageDataGenerator(
+            featurewise_center=False,
+            samplewise_center=False,
+            featurewise_std_normalization=False,
+            samplewise_std_normalization=False,
+            zca_whitening=False,
+            rotation_range=0,
+            width_shift_range=0,
+            height_shift_range=0,
+            shear_range=0,
+            horizontal_flip=True,
+            vertical_flip=False,
+            dim_ordering=dim_ordering).flow(x, [1])
+        for i in range(10):
```

Comment:
I use the for loop here only to increase the likelihood that an image is flipped.  If anyone suggests a more graceful way to do this, I'll clean it up.

Similarly, I wasn't sure how to set a seed in this test to ensure that random_transform always performs the same  operations during this test.  The previous test in this file allows some randomness in the random_transform operation, so I'm guessing that's acceptable.  If we want these tests to be deterministic, and know how to make that happen, I'm happy to fix them both.